gpytorch/variational/nearest_neighbor_variational_strategy.py

```diff
@@ -359,5 +434,7 @@ def _compute_nn(self) -> "NNVariationalStrategy":
         with torch.no_grad():
             inducing_points_fl = self.inducing_points.data.float()
             self.nn_util.set_nn_idx(inducing_points_fl)
-            self.nn_xinduce_idx = self.nn_util.build_sequential_nn_idx(inducing_points_fl)
+            if self.k < self.M:
+                self.nn_xinduce_idx = self.nn_util.build_sequential_nn_idx(inducing_points_fl)
```

Comment:
Do we have a test in code for the `k < M` case? If not, can you add one?

Reply:
Yeah I think k = 3 in the test code, which is smaller than M. I could add a case when k = M later, which is missing currently.